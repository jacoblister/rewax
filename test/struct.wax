(struct empty
  (let dummy int)
)

(func struct_empty_test
  (print "struct_empty_test:")
  (let e (struct empty) (alloc (struct empty)))
)

(struct point 
    (let pos_x float)
    (let pos_y float)
)

(func struct_simple_test
  (print "struct_simple_test:")
  (let p (struct point) (alloc (struct point)))
  (set p pos_x 1.1)
  (set p pos_y 1.2)
  (print (get p pos_x))
  (print (get p pos_y))
  (free p)
)

(struct node 
    (let name str)
    (let child (struct node))
)

(func struct_nested_test
  (print "struct_nested_test:")
  (let parent (struct node) (alloc (struct node)))
  (set parent name "parent")
  
  (let child (struct node) (alloc (struct node)))
  (set child name "child")

  (set parent child child)

  (print (get parent name))
  (print (get parent child name))

  (free child)
  (free parent)
)

(func struct_comparison_value_test
  (print "struct_comparison_value_test:")
  (let a (struct node) (alloc (struct node)))
  (set a name "A")
  (let b (struct node) (alloc (struct node)))
  (set b name "A")
  (print (get a name))
  (print (get b name))
  (print (+ (= a b) 0))
)

(func struct_comparison_ref_test
  (print "struct_comparison_ref_test:")
  (let a (struct node) (alloc (struct node)))
  (set a name "A")
  (let b (struct node) a)
  (print (get a name))
  (print (get b name))
  (print (+ (= a b) 0))
)

(func main (result int)
  (call struct_empty_test)
  (call struct_simple_test)
  (call struct_nested_test)
  (call struct_comparison_value_test)
  (call struct_comparison_ref_test)
  (return 0)
)