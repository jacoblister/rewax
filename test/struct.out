struct_empty_test:
struct_simple_test:
1.1
1.2
struct_nested_test:
parent
child
struct_comparison_value_test:
A
A
0
struct_comparison_ref_test:
A
A
1
